### DISCUSSION TITLE
Error: The Serverless Function "index.js" is 52.27mb which exceeds the maximum size limit of 50mb. Learn More: https://vercel.link/serverless-function-size

### DISCUSSION THREAD

**Discussion by eternalodball029** [vercel/vercel#5532] [Help] [answered]
Is there any way or hack to troubleshoot this error .............

**Comment by jpbecotte** [comment]
This is a bit annoying! Getting the same error and trying to figure out why!

**Reply by thmsnhl** [reply]
Did you manage to resolve the issue?

**Reply by jpbecotte** [reply]
Yes. Somehow the node_modules directory is too big. Most packages must be moved to dev dependencies to only generate the final build and you have to figure out which one of them have to be moved.

**Comment by NoahBres** [comment]
Are there any documented steps into troubleshooting this error?

I am running into this error as vercel throws an error stating that my serverless functions is 108mb!! The function in question is 115 lines long with 2 dependencies, one of which is 2.6kb gzipped and the other 12. There is no way this function alone is 108mb.

How is next/vercel bundling these functions? What's being included that balloons this function size? I can't seem to find any documentation on this.

I've included the following in `vercel.json` in the event that the static files were being included. But this does not help at all.
```json
{
  "functions": {
    "pages/api/contact.ts": {
      "excludeFiles": "public/**/*"
    }
  }
}
```

**Reply by NoahBres** [reply]
My issue seems to be related to these:

https://github.com/vercel/vercel/discussions/6366
https://github.com/vercel/next.js/issues/26297

**Comment by leerob** [comment, marked as answer]
Replaced by https://github.com/vercel/vercel/discussions/6516, please continue the discussion there 🙏
